packages/bugsnag_flutter_performance/lib/src/span_control/prioritized_store.dart

```diff
@@ -0,0 +1,63 @@
+typedef AddBlock<T> = void Function(T object, int priority);
+typedef BatchBlock<T> = void Function(AddBlock<T> addBlock);
+
+class PrioritizedStoreEntry<T> {
+  final T object;
+  final int priority;
+
+  PrioritizedStoreEntry(this.object, this.priority);
+}
+
+class PrioritizedStore<T> {
+  final List<PrioritizedStoreEntry<T>> _store = [];
+  final Set<T> _uniqueObjects = {};
+  List<T> _objects = [];
+
+  List<T> get objects => List.unmodifiable(_objects);
```

Comment:
Could this be an `Iterable<T>` instead of a `List`? That way the `map` would remain cold and avoid the overhead of maintaining the `_objects` state.

```suggestion
  Iterable<T> get objects => _store.map((entry) => entry.object);
```